### PR TITLE
Add common components

### PR DESCRIPTION
**Pull request by morganee**
Add 2 properties to Modal
Add new component FormInputWrapper that mimic the "ValidXX" ones that we have to be generic
Add Icon, TooltipIcon, TablePaginationFooter, Tabs
Updates design system version

### CODE REVIEW

**Comment by mnicole** [comment]
thanks for the updates!! :) 
just had a few comments.. the main one is the build is failing since gumdrops doesn't have a dependency on react router

**Comment by mnicole** [comment]
oh also, i think you need to import the new components in components/index.js too to have them build properly when we build the library

**Comment by morganee** [comment]
> oh also, i think you need to import the new components in components/index.js too to have them build properly when we build the library

It fixed it ! thanks. I removed the <Link> option in the Tabs component too, to remove the react-router-dom dependency, we can use the onclick function and it's cleaner I think

**Comment by mnicole** [comment]
thanks, the branch built now! are you going to address the other comments too? :)

**Comment by morganee** [comment]
> thanks, the branch built now! are you going to address the other comments too? :)

Have I missed one ? I'm only seeing the one about the react-router which I removed

**Review by mnicole — CHANGES_REQUESTED** [review]

**Comment by mnicole** [comment]
oh oops the UI is confusing... i think i had to "Submit" my review to have you see all the comments.. do you see them now?

**Comment by morganee** [comment]
> oh oops the UI is confusing... i think i had to "Submit" my review to have you see all the comments.. do you see them now?

yes ! thanks 👍

File: components/atoms/FormInputWrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FormGroup from '../molecules/FormGroup';
+import FormGroupLabel from './FormGroupLabel';
+import FormGroupTextHelp from './FormGroupTextHelp';
+
+const DEFAULT_STYLES = {
+    position: 'absolute',
```
[the remark this answers is not in the document]
Comment (morganee) [reply components/atoms/FormInputWrapper.jsx:9]: To place the helper text right below the input. We add them by default to all the ValidXX component so I added them here too

File: components/atoms/Icon.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const contextToClassName = {
+    primary: '-color-tx-pri-lt-4',
+    secondary: '-color-tx-war-lt-4',
+    danger: '-color-tx-dan-lt-4',
+};
+
+const Icon = ({ className, icon, prefix, fontSize, context, size, ...rest }) => {
```
[the remark this answers is not in the document]
Comment (morganee) [reply components/atoms/Icon.jsx:10]: It's not showing ? I added the mdx and stories file and the index file

File: components/molecules/TablePaginationFooter.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            )}
+        </Fragment>
+    );
+};
+
+TablePaginationFooter.propTypes = {
+    totalPages: PropTypes.number.isRequired,
+    totalElements: PropTypes.number.isRequired,
+    handlePageChange: PropTypes.func.isRequired,
+    handleSizeChange: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (morganee) [reply components/molecules/TablePaginationFooter.jsx:62]: yes thanks I'll update

File: _stories/molecules/TablePaginationFooter/TablePaginationFooter.mdx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Story, Props } from '@storybook/addon-docs/blocks';
+import TablePaginationFooter from '../../../components/molecules/TablePaginationFooter';
+
+# TablePaginationFooter
+
+`<TablePaginationFooter>` extends `<Pagination>` to add a size dropwown to control the number of element displayed per page in the associated table.
+
+`handlePageChange` and `handleSizeChange` are used to update the table data to reflect the pagination state.
```
[the remark this answers is not in the document]
Comment (morganee) [reply _stories/molecules/TablePaginationFooter/TablePaginationFooter.mdx:8]: done

File: components/atoms/Icon.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnicole) [reply components/atoms/Icon.jsx:10]: it has ToolTipIcon but not Icon... btw you can see the build here https://deploy-preview-104--gumdrops.netlify.app/
anyway it's fine to finish this after your vacation :)

[the remark this answers is not in the document]
Comment (morganee) [reply components/atoms/Icon.jsx:10]: It's added. They were ignored by git because Icon was in the .gitignore as a default for Mac's

**Review by mnicole — APPROVED** [review]

**Comment by garrettbear** [comment]
Sorry I just saw your updated commit and all looks good. Thanks for doing this!

**Pull request merged**